pkg/kubelet/apis/podresources/server_v1.go

```diff
@@ -70,3 +70,45 @@ func (p *v1PodResourcesServer) List(ctx context.Context, req *v1.ListPodResource
 		PodResources: podResources,
 	}, nil
 }
+
+// GetAllocatableResources returns information about all the devices known by the server
+func (p *v1PodResourcesServer) GetAllocatableResources(ctx context.Context, req *v1.AllocatableResourcesRequest) (*v1.AllocatableResourcesResponse, error) {
+	metrics.PodResourcesEndpointRequestsTotalCount.WithLabelValues("v1").Inc()
+
+	allDevices := p.devicesProvider.GetAllDevices()
+	var respDevs []*v1.ContainerDevices
+
+	for resourceName, resourceDevs := range allDevices {
+		for devID, dev := range resourceDevs {
+			topo := dev.GetTopology()
+			if topo == nil {
+				// Some device plugin do not report the topology information.
+				// This is legal, so we report the devices anyway,
+				// let the client decide what to do.
+				respDevs = append(respDevs, &v1.ContainerDevices{
+					ResourceName: resourceName,
+					DeviceIds:    []string{devID},
+				})
+				continue
+			}
+
+			// we expect a single node anyway, but this deals with nil gracefully
+			for _, node := range topo.GetNodes() {
+				respDevs = append(respDevs, &v1.ContainerDevices{
+					ResourceName: resourceName,
+					DeviceIds:    []string{devID},
+					Topology: &v1.TopologyInfo{
+						Nodes: []*v1.NUMANode{
+							{ID: node.GetID()},
```

Comment:
it can be multiple ids per device, as announced by device plugin.

Reply:
ack, I need to fix this.